Server/ReasnAPI/ReasnAPI/Services/UserService.cs

```diff
@@ -1,16 +1,292 @@
+
+                if (user is null)
+                {
+                    return null;
+                }
+
+                var usernameExists = context.Users.Any(r => r.Username == userDto.Username && r.Id != userId);
+
+                if (usernameExists)
+                {
+                    return null;
+                }
+
+                user.Username = userDto.Username;
+                user.Name = userDto.Name;
+                user.Surname = userDto.Surname;
+                user.Username = userDto.Username;
+                user.Email = userDto.Email;
+                user.Phone = userDto.Phone;
+                user.Role = userDto.Role;
+                user.AddressId = userDto.AddressId;
+                user.UpdatedAt = DateTime.UtcNow;
+
+                context.Users.Update(user);
+
+                if (userDto.Interests is null || userDto.Interests.Count == 0)
+                {
+                    context.SaveChanges();
+                    return MapToUserDto(user);
+                }
+
+                var userInterestsDb = context.UserInterests
+                                           .Where(ui => ui.UserId == userId)
+                                           .ToList();
+
+                foreach (var userInterest in userInterestsDb)
+                {
+                    var interestDb = context.Interests.Find(userInterest.InterestId);
+
+                    if (interestDb is null)
+                    {
+                        continue;
+                    }
+
+                    if (userDto.Interests.Exists(i => i.Interest.Name == interestDb.Name))
+                    {
+                        var userInterestDto = userDto.Interests.Find(i => i.Interest.Name == interestDb.Name);   
+
+                        if (userInterestDto is null)
+                        {
+                            continue;
+                        }
+
+                        userInterest.Level = userInterestDto.Level;
+                        context.UserInterests.Update(userInterest);
+                        userDto.Interests.RemoveAll(i => i.Interest.Name == interestDb.Name);
+                    }
+                    else
+                    {
+                        context.UserInterests.Remove(userInterest);
+                    }
+                }
+
+                foreach (var interest in userDto.Interests)
+                {
+                    var interestDb = context.Interests.FirstOrDefault(i => i.Name == interest.Interest.Name);
+
+                    if (interestDb is null)
+                    {
+                        continue;
+                    }
+
+                    var newInterest = new UserInterest
+                    {
+                        UserId = user.Id,
+                        InterestId = interestDb.Id,
+                        Level = interest.Level
+                    };
+
+                    context.UserInterests.Add(newInterest);
+                }

```

Comment:
I think we can simply get via linq all interests that:

1. are in db, are not in dto - then removerange
2. are in dto, are not in db - then addrange
3. are in dto and db but have different levels - updaterange